File: frameworks/compatible/components/stepper/stepper_item_modifier.cpp
/*
 * Copyright (c) 2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "compatible/components/stepper/stepper_modifier_api.h"
#include "frameworks/compatible/components/stepper/stepper_item_component.h"
#include "interfaces/inner_api/ace_kit/include/ui/base/ace_type.h"

namespace OHOS::Ace::StepperItemModifier {

RefPtr<Component> CreateStepperItem()
{
    return AceType::MakeRefPtr<StepperItemComponent>(RefPtr<Component>());
}

RefPtr<Component> UpdateStepperItem(const RefPtr<Component>& component)
{
    return AceType::DynamicCast<StepperItemComponent>(component);
}

const ArkUIStepperItemComponentModifier* GetStepperItemComponentModifier()
{
    static const ArkUIStepperItemComponentModifier instance = {
        .createStepperItem = CreateStepperItem,
        .updateStepperItem = UpdateStepperItem,
    };
    return &instance;
}
} // OHOS::Ace::StepperItemModifier

File: frameworks/bridge/declarative_frontend/ark_modifier/src/stepper_item_modifier.ts
/*
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

class LazyArkStepperItemComponent extends ArkComponent {
  static module: StepperItemComponentModule | undefined = undefined;
  constructor(nativePtr: KNode, classType: ModifierType) {
   super(nativePtr, classType);
   if (LazyArkStepperItemComponent.module === undefined) {
     LazyArkStepperItemComponent.module = globalThis.requireNapi('arkui.components.arkstepperitem');
   }

   this.lazyComponent = LazyArkStepperItemComponent.module.createComponent(nativePtr, classType);
  }

  setMap(): void {
   this.lazyComponent._modifiersWithKeys = this._modifiersWithKeys;
  }

  prevLabel(value: string): this {
   this.lazyComponent.prevLabel(value);
   return this;
  }

  nextLabel(value: string): this {
   this.lazyComponent.nextLabel(value);
   return this;
  }

  status(value?: ItemState): this {
   this.lazyComponent.status(value);
   return this;
  }
}

class StepperItemModifier extends LazyArkStepperItemComponent implements AttributeModifier<StepperItemAttribute> {

  constructor(nativePtr: KNode, classType: ModifierType) {
    super(nativePtr, classType);
    this._modifiersWithKeys = new ModifierMap();
    this.setMap();
  }

  applyNormalAttribute(instance: StepperItemAttribute): void {
    ModifierUtils.applySetOnChange(this);
    ModifierUtils.applyAndMergeModifier<StepperItemAttribute, ArkStepperItemComponent, ArkComponent>(instance, this);
  }
}
